### PR TITLE
Remove unneeded Microsoft/CMake specific code

### PR DESCRIPTION
**Pull request by jgoshi**
I removed as much code as I could so that our fork is as similar to Kitware master as possible. The main things I removed were:
1. VCSystemIncludePaths - This was no longer needed because we have environment variables that server the same purpose.

2. Removed code to find VS 2017. I am now using the same code as is in the Kitware master branch. To make sure it worked I went into the debugger and checked that the code for finding VS2017 is working.

### CODE REVIEW

File: Source/cmGlobalVisualStudio15Generator.cxx [hunk 1]
```diff
@@ -172,75 +171,36 @@ bool cmGlobalVisualStudio15Generator::IsWin81SDKInstalled() const
...
-      return DevEnvPath;
-    }
-    else {
-      return "";
+  std::string devenv;
+
+  // Ask Visual Studio Installer tool.
+  std::string vs;
+  if (vsSetupAPIHelper.GetVSInstanceInfo(vs)) {
+    devenv = vs + "/Common7/IDE/devenv.com";
```
Comment (rchiodo) [inline Source/cmGlobalVisualStudio15Generator.cxx:198]: Devenv.com? Why did you change this part?

Comment (jgoshi) [reply Source/cmGlobalVisualStudio15Generator.cxx:198]: Because this is what is in Kitware. Unless we have to change it it's much easier to just keep what they have.

Comment (rchiodo) [reply Source/cmGlobalVisualStudio15Generator.cxx:198]: Okay sounds good. Thanks

**Review by rchiodo — APPROVED** [review]
